commit 27d50646d0815f02677ff870a5691f36be67c08f
Author: Justin Stitt <justinstitt@google.com>
Date:   Mon Mar 18 23:46:33 2024 +0000

    fbdev: au1200fb: replace deprecated strncpy with strscpy
    
    strncpy() is deprecated for use on NUL-terminated destination strings
    [1] and as such we should prefer more robust and less ambiguous string
    interfaces.
    
    Let's use the new 2-argument strscpy() which guarantees NUL-termination
    on the destination buffer while also simplifying the syntax. Note that
    strscpy() will not NUL-pad the destination buffer like strncpy() does.
    
    However, the NUL-padding behavior of strncpy() is not required since
    fbdev is already NUL-allocated from au1200fb_drv_probe() ->
    frameuffer_alloc(), rendering any additional NUL-padding redundant.
    |       p = kzalloc(fb_info_size + size, GFP_KERNEL);
    
    Link: https://www.kernel.org/doc/html/latest/process/deprecated.html#strncpy-on-nul-terminated-strings [1]
    Link: https://manpages.debian.org/testing/linux-manual-4.8/strscpy.9.en.html [2]
    Link: https://github.com/KSPP/linux/issues/90
    Cc: linux-hardening@vger.kernel.org
    Signed-off-by: Justin Stitt <justinstitt@google.com>
    Reviewed-by: Kees Cook <keescook@chromium.org>
    Signed-off-by: Helge Deller <deller@gmx.de>

diff --git a/drivers/video/fbdev/au1200fb.c b/drivers/video/fbdev/au1200fb.c
index 6f20efc663d7..e718fea63662 100644
--- a/drivers/video/fbdev/au1200fb.c
+++ b/drivers/video/fbdev/au1200fb.c
@@ -1557,7 +1557,7 @@ static int au1200fb_init_fbinfo(struct au1200fb_device *fbdev)
 		return ret;
 	}
 
-	strncpy(fbi->fix.id, "AU1200", sizeof(fbi->fix.id));
+	strscpy(fbi->fix.id, "AU1200");
 	fbi->fix.smem_start = fbdev->fb_phys;
 	fbi->fix.smem_len = fbdev->fb_len;
 	fbi->fix.type = FB_TYPE_PACKED_PIXELS;